src/main/java/com/wire/helium/API.java

```diff
@@ -618,6 +628,273 @@ public User getSelf() {
+                .header(HttpHeaders.AUTHORIZATION, bearer(token))
+                .put(Entity.json(clientUpdate));
+
+            if (response.getStatus() >= 400) {
+                Logger.error("uploadClientPublicKey error, bad-request");
+            } else if (response.getStatus() == 200) {
+                Logger.info("uploadClientPublicKey success");
+            }
+        } catch (Exception e) {
+            Logger.error("uploadClientPublicKey error: %s", e.getMessage());
+        }
+    }
+
+    /**
+     * <p>
+     *     To upload client key packages we POST a {@link KeyPackageUpdate} object containing a list of package keys
+     *     to /mls/key-packages/self/{clientId}
+     * </p>
+     * <p>
+     *     As there is no return, in case it fails we just map the HTTP Code and log the message.
+     * </p>
+     *
+     * @param clientId          clientId to upload the package keys
+     * @param keyPackageUpdate  list of package keys
+     */
+    @Override
+    public void uploadClientKeyPackages(String clientId, KeyPackageUpdate keyPackageUpdate) {
+        try {
+            Response response = mlsPath
+                .path("key-packages")
+                .path("self")
+                .path(clientId)
+                .request(MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.AUTHORIZATION, bearer(token))
+                .post(Entity.json(keyPackageUpdate));
+
+            if (response.getStatus() == 403) {
+                Logger.error("uploadClientKeyPackages error, mls-identity-mismatch");
+            } else if (response.getStatus() == 400) {
+                Logger.error("uploadClientKeyPackages error, mls-protocol-error");
+            } else if (response.getStatus() == 200) {
+                Logger.error("uploadClientKeyPackages success");
+            }
+        } catch (Exception e) {
+            Logger.error("uploadClientKeyPackages error: %s", e.getMessage());
+        }
+
+    }
+
+    @Override
+    public byte[] getConversationGroupInfo(QualifiedId conversationId) {
+        try {
+            Response response = conversationsPath
+                .path(conversationId.domain)
+                .path(conversationId.id.toString())
+                .path("groupinfo")
+                .request(MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.AUTHORIZATION, bearer(token))
+                .accept("message/mls")
+                .get();
+
+            if (response.getStatus() == 200) {
+                return response.readEntity(byte[].class);
+            }
+
+            if (response.getStatus() >= 400) {
+                String msgError = response.readEntity(String.class);
+                Logger.error("getConversationGroupInfo error: %s, status: %d", msgError, response.getStatus());
+            }
+        } catch (Exception e) {
+            Logger.error("getConversationGroupInfo error: %s", e.getMessage());
+        }
+        return new byte[0];
```

Comment:
Not sure that if the request fails, it should return silently a wrong value.
if response.getStatus() != 2XX then log the error but probably throw an exception so LegalHold knows it failed

Reply:
Added a throw of `RuntimeException` containing a `HttpException` with response values